src/notification/toast-notification.component.js

```diff
@@ -176,69 +176,78 @@
 
  </example>
  */
-angular.module( 'patternfly.notification' ).directive('pfToastNotification', function () {
-  'use strict';
-
-  return {
-    scope: {
-      'notificationType': '@',
-      'message': '@',
-      'header': '@',
-      'showClose': '@',
-      'closeCallback': '=?',
-      'actionTitle': '@',
-      'actionCallback': '=?',
-      'menuActions': '=?',
-      'updateViewing': '=?',
-      'data': '=?'
-    },
-    restrict: 'A',
-    templateUrl: 'notification/toast-notification.html',
-    controller: function ($scope) {
-      $scope.notificationType = $scope.notificationType || 'info';
-
-      $scope.updateShowClose = function () {
-        $scope.showCloseButton = ($scope.showClose === 'true') && (angular.isUndefined($scope.menuActions) || $scope.menuActions.length < 1);
-      };
-
-      $scope.handleClose = function () {
-        if (angular.isFunction($scope.closeCallback)) {
-          $scope.closeCallback($scope.data);
-        }
-      };
-
-      $scope.handleAction = function () {
-        if (angular.isFunction($scope.actionCallback)) {
-          $scope.actionCallback($scope.data);
-        }
-      };
-
-      $scope.handleMenuAction = function (menuAction) {
-        if (menuAction && angular.isFunction(menuAction.actionFn) && (menuAction.isDisabled !== true)) {
-          menuAction.actionFn(menuAction, $scope.data);
-        }
-      };
-
-      $scope.handleEnter = function () {
-        if (angular.isFunction($scope.updateViewing)) {
-          $scope.updateViewing(true, $scope.data);
-        }
-      };
-      $scope.handleLeave = function () {
-        if (angular.isFunction($scope.updateViewing)) {
-          $scope.updateViewing(false, $scope.data);
-        }
-      };
-
-      $scope.updateShowClose ();
-    },
-    link: function (scope) {
-      scope.$watch('showClose', function () {
-        scope.updateShowClose();
-      });
-      scope.$watch('menuActions', function () {
-        scope.updateShowClose();
-      });
-    }
-  };
+angular.module( 'patternfly.notification' ).component('pfToastNotification', {
+  bindings: {
+    'notificationType': '@',
+    'message': '@',
+    'header': '@',
+    'showClose': '@',
+    'closeCallback': '=?',
+    'actionTitle': '@',
+    'actionCallback': '=?',
+    'menuActions': '<?',
+    'updateViewing': '=?',
+    'data': '=?'
+  },
+  templateUrl: 'notification/toast-notification.html',
+  controller: function () {
+    'use strict';
+    var ctrl = this,
+      _showClose;
+
+    Object.defineProperty(ctrl, 'showClose', {
+      get: function () {
+        return _showClose;
+      },
+      set: function (value) {
+        _showClose = value;
+        ctrl.updateShowClose();
+      }
+    });
+
```

Comment:
We'd have to change showClose from @ to < to do that, I was trying to leave the interface the same so it can use {{ }}.  I don't mind changing and documenting if you prefer @dtaylor113